src/dataflow/src/decode/mod.rs

```diff
@@ -536,84 +531,71 @@ where
                         &value_buf
                     };
 
-                    if value.is_empty() {
+                    // At this point, `value` is non-empty, as it came from a
```

Comment:
Does the code care if it's non-empty?

Reply:
No, but the old code clearly did something else (but I think was never called?) 

If you think its fine to send `Some("")` with an actual position, then I can remove the `MessagePayload::Data` must be non-empty invariant, but I have no idea how the rest of the the code, including `sql`, will react if we send it empty data, so I am inclined to be defensive

Reply:
I think we should either make this an invariant checked by a `MessagePayload::new` constructor to see a panic if we ever violate it, or not make it an invariant at all. My understanding is that no part of the code cares about the data being empty as we have a separate variant to communicate EOF so I'd lean towards the latter